wasm-engines.md

```diff
@@ -0,0 +1,224 @@
+# Wasm Engines
+
+Binaryen and Wabt are two WebAssembly Engines which also provides a set of tools
+for compiling and decompiling wasm binary files. Both are official WebAssembly Projects.
+
+# Binaryen
+
+## Getting and compiling binaryen
+
+Install development tools (`build-essential`) `cmake` and `make`:
+
+```
+sudo apt-get install build-essential cmake make
+```
+
+Clone the official Binaryen repository:
+
+```
+git clone https://github.com/WebAssembly/binaryen.git
+```
+
+Move to the new `binaryen` directory and run this command to build the tools:
+
+```
+cmake . && make
+```
+
+Now you have the following binaryen tools compiled in the `bin/` directory:
+
+- **asm2wasm**: An asm.js to WebAssembly compiler.
```

Comment:
This is very thorough! But instead of linking to the individual tools here, it's probably better to just link to https://github.com/WebAssembly/binaryen#tools. For the purposes of Ewasm, users will only be using the three tools that you describe in more detail in the next section, anyway.